src/main/java/com/lalabrand/ecommerce/item/category/CategoryService.java

```diff
@@ -1,36 +1,52 @@
 package com.lalabrand.ecommerce.item.category;
 
+import com.lalabrand.ecommerce.user.wishlist.WishlistService;
+import jakarta.persistence.EntityNotFoundException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 import org.springframework.stereotype.Service;
 
 import java.util.Collections;
 import java.util.List;
 import java.util.Optional;
+import java.util.stream.Collectors;
 
 
 @Service
 public class CategoryService {
     private final CategoryRepository categoryRepository;
+    private final Logger logger = LoggerFactory.getLogger(CategoryService.class);
 
     public CategoryService(CategoryRepository categoryRepository) {
         this.categoryRepository = categoryRepository;
     }
 
     public List<CategoryDTO> findAllCategory() {
         try {
-            return categoryRepository.findAll().stream().map(CategoryDTO::fromEntity).toList();
-        } catch (Exception e) {
-            return Collections.emptyList();
+            List<Category> categories = categoryRepository.findAll();
+            if (categories.isEmpty()) {
+                throw new EntityNotFoundException("No categories found");
+            }
+            return categories.stream()
+                    .map(CategoryDTO::fromEntity)
+                    .collect(Collectors.toList());
+        } catch (EntityNotFoundException ex) {
+            logger.info("No categories found: " + ex.getMessage());
+            throw ex;
+        } catch (Exception ex) {
+            logger.info("Error occurred while fetching categories: " + ex.getMessage());
+            throw new RuntimeException("Error occurred while fetching categories", ex);
         }
     }
 
     public Optional<CategoryDTO> findCategoryById(Integer id) {
-        try {
-            if (id == null || id < 1) {
-                throw new IllegalArgumentException("Id can not be null or less than 1");
-            }
-            return categoryRepository.findById(id).map(CategoryDTO::fromEntity);
-        } catch (IllegalArgumentException illegalArgumentException) {
-            return Optional.empty();
+        if (id == null || id < 1) {
```

Comment:
@denomelchenko Why `id` cannot be 0? In src/main/java/com/lalabrand/ecommerce/item/ItemService.java#33 you check for 0.

Reply:
Because in MySQL, auto-incrementing starts with 1